connect/src/routes/tokenBridgeRoute.ts

```diff
@@ -0,0 +1,87 @@
+import { Network } from "@wormhole-foundation/sdk-base";
+import {
+  Signer,
+  TokenTransferDetails,
+  TransactionId,
+  isSameToken,
+} from "@wormhole-foundation/sdk-definitions";
+import { Wormhole } from "../wormhole";
+import { Route, TransferRequest, ValidationError, ValidationResult } from "./route";
+import { TokenTransfer } from "../protocols/tokenTransfer";
+
+export type TokenBridgeTransferOptions = {
+  payload?: Uint8Array;
+};
+
+export class TokenBridgeRoute<N extends Network> extends Route<N, TokenBridgeTransferOptions> {
+  NATIVE_GAS_DROPOFF_SUPPORTED = false;
+  IS_AUTOMATIC = false;
+
+  constructor(wh: Wormhole<N>, request: TransferRequest) {
+    super(wh, request);
+  }
+
+  private toTransferDetails(options: TokenBridgeTransferOptions): TokenTransferDetails {
+    return {
+      token: this.request.source,
+      amount: this.request.amount,
+      from: this.request.from,
+      to: this.request.to,
+      ...options,
+    };
+  }
+
+  async isSupported(): Promise<boolean> {
+    // No transfers to same chain
+    if (this.request.from.chain === this.request.to.chain) return false;
+
+    // No transfers to unsupported chains
+    const fromChain = this.wh.getChain(this.request.from.chain);
+    if (!fromChain.supportsTokenBridge()) return false;
+
+    const toChain = this.wh.getChain(this.request.to.chain);
+    if (!toChain.supportsTokenBridge()) return false;
+
+    // If the destination token was set, and its different than what
+    // we'd get from a token bridge transfer, then this route is not supported
+    if (this.request.destination) {
+      const destToken = await TokenTransfer.lookupDestinationToken(
+        fromChain,
+        toChain,
+        this.request.source,
+      );
+      if (!isSameToken(destToken, this.request.destination)) return false;
```

Comment:
Connect's code has all kinds of special cases I don't understand. I think we need it in this SDK too.

https://github.com/wormhole-foundation/wormhole-connect/blob/development/wormhole-connect/src/routes/bridge/bridge.ts#L48-L62